subsys/net/ip/tcp.c

```diff
@@ -4043,6 +4043,20 @@ int net_tcp_finalize(struct net_pkt *pkt, bool force_chksum)
 {
 	NET_PKT_DATA_ACCESS_DEFINE(tcp_access, struct net_tcp_hdr);
 	struct net_tcp_hdr *tcp_hdr;
+	bool is_ipv6;
```

Comment:
This could be simplified to a single line:
```c
enum net_if_checksum_type type = net_pkt_family(pkt) == AF_INET6 ?  
	NET_IF_CHECKSUM_IPV6_TCP : NET_IF_CHECKSUM_IPV4_TCP;
```
and the use `type` as an argument in the function. Same in similar other places.

Reply:
Good idea, done.